eppo_client/client.py

```diff
@@ -286,102 +288,112 @@ def get_bandit_action(
             },
             "default"
         )
-        if result.action is None:
-            do_variation(result.variation)
+        if result.action:
+            do_action(result.variation)
         else:
-            do_action(result.action)
+            do_status_quo()
         """
+        variation = default
+        action = None
         try:
-            return self.get_bandit_action_detail(
+            subject_attributes = convert_context_attributes_to_attributes(subject_context)
+
+            # get experiment assignment
+            # ignoring type because Dict[str, str] satisfies Dict[str, str | ...] but mypy does not understand
+            variation = self.get_string_assignment(
                 flag_key,
                 subject_key,
-                subject_context,
-                actions,
+                subject_attributes,
                 default,
             )
+
+            if variation in self.get_bandit_keys():
+                action = self.evaluate_bandit_action(
+                    flag_key,
+                    variation, # for now, we assume the variation value is always equal to the bandit key
+                    subject_key,
+                    subject_context,
+                    actions,
+                )
         except Exception as e:
             if self.__is_graceful_mode:
                 logger.error("[Eppo SDK] Error getting bandit action: " + str(e))
-                return BanditResult(default, None)
-            raise e
+            else:
+                raise e
+
+        return BanditResult(variation, action)
 
-    def get_bandit_action_detail(
+    def evaluate_bandit_action(
```

Comment:
Moved assignment getting upstream, so now this method's only responsibility is to get and log the selected action